src/hermes_claude_agent_sdk/parity/inventory.py

```
@@ -0,0 +1,379 @@
+def build_declared_inventory_from_tool_schemas(
+    tool_schemas: Sequence[Any], *, declared_by: str = "host", enabled: bool = True,
+    mcp_servers: Sequence[MCPServerInventoryEntry | Mapping[str, Any]] = (),
+) -> DeclaredInventory:
+    if not isinstance(declared_by, str) or declared_by not in _SIDES:
+        raise InventoryValidationError("declared_by must be host or plugin")
+    active = _bool(enabled)
+    try:
+        from ..tool_bridge import normalize_tool_schemas
+        definitions = normalize_tool_schemas(tool_schemas)
+    except Exception as exc:
+        raise InventoryValidationError("public tool schema normalization failed") from exc
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)